build.gradle

```diff
@@ -107,8 +107,16 @@ allprojects {
     }
 
     tasks.withType(JavaCompile) {
-        // temporarily relax constraints until we can fix all violations
-        options.errorprone.warn 'CatchBlockLogException',
+        options.errorprone {
+            disableWarningsInGeneratedCode = true
+            excludedPaths = '.*(generated).*'
+
+            // increase specific warnings to error until we can treat all warnings as errors
+            error 'BoxedPrimitiveEquality',
+                'ReferenceEquality'
```

Comment:
These are not technically required with gradle-baseline 3.43.0+ per https://github.com/palantir/gradle-baseline/pull/1514 so curious if folks want to remove them from here or leave them to be explicit in case gradle-baseline relaxes either back to a warning.